plugins/de.cau.cs.kieler.spviz.spviz/src/de/cau/cs/kieler/spviz/spviz/generator/GenerateSyntheses.xtend

```diff
@@ -0,0 +1,1225 @@
+						usedContext.setProperty(SynthesisProperties.CURRENT_VISUALIZATION_CONTEXT_INDEX, index)
+						visualizationContext = VizModelUtil.create«spviz.vizName»(model)
+						visualizationContexts.add(visualizationContext)
+					}
+			
+					// Make this variable final here for later usage in the lambda.
+					val visContext = visualizationContext
+					if (visContext.focus === null) {
+						// This synthesis can be used.
+						
+						// The overview of the entire OSGi Project.
+						modelNode.children += createNode => [
+							associateWith(model)
+							data += createKIdentifier => [ it.id = visContext.hashCode.toString ]
+							SynthesisUtils.configureBoxLayout(it)
+							setLayoutOption(BoxLayouterOptions.BOX_PACKING_MODE, PackingMode.GROUP_MIXED)
+							addProjectRendering
+							«FOR overview : spviz.overviews»
+								
+								val overview«overview»Nodes = «overview.toFirstLower»OverviewSynthesis.transform(visContext.«overview.toFirstLower»OverviewContext)
+								children += overview«overview»Nodes
+							«ENDFOR»
+						]
+						
+						return modelNode
+						
+					} else {
+						// Delegate the view model generation to another subsynthesis that can show the requested visualization context.
+						val children = transformSubModel(visualizationContext.focus)
+						
+						modelNode.children += children
+						
+						return modelNode
+					}
+				}
+				
+				private def transformSubModel(IVisualizationContext<?> context) {
+					switch (context) {
+						«FOR overview : spviz.overviews»
+							«overview»OverviewContext: {
+								return «overview.toFirstLower»OverviewSynthesis.transform(context)
+							}
+						«ENDFOR»
+						default: {
+							throw new IllegalArgumentException("The context class has no known subsynthesis: " + context.class)
+						}
+					}
+				}
+				
+			}
+		'''
+	}
+	
+	/**
+	 * Generates the contend for the klighd setup.
+	 * 
+	 * @param spviz
+	 * 		a DataAccess to easily get the information from
+	 * @return 
+	 * 		the generated file content as a string
+	 */
+	def static String generateKlighdSetup(DataAccess spviz){
+		return '''
+			package «spviz.packageName».viz
+			
+			import de.cau.cs.kieler.klighd.IKlighdStartupHook
+			import de.cau.cs.kieler.klighd.KlighdDataManager
+«««			import «spviz.packageName».viz.actions.ConnectAllAction
+			import «spviz.packageName».viz.actions.ContextCollapseExpandAction
+«««			import «spviz.packageName».viz.actions.ContextExpandAllAction
+«««			import «spviz.packageName».viz.actions.ContextRemoveAction
+«««			import «spviz.packageName».viz.actions.FocusAction
+			import «spviz.packageName».viz.actions.OverviewContextCollapseExpandAction
+			«FOR overview : spviz.overviews»
+			«FOR connection : spviz.getOverviewConnections(overview)»
+				import «spviz.packageName».viz.actions.RevealRequired«connection.get(0)»«connection.get(2)»sAction
+				import «spviz.packageName».viz.actions.RevealRequiring«connection.get(0)»«connection.get(1)»sAction
```

Comment:
I see these `get(0)`, `get(1)`, etc. calls often here. This is not really speaking what these are doing. Having an own class for the connection with better named field accesses would make the code more readable.